bugwarrior/services/__init__.py

```diff
@@ -165,6 +171,8 @@ def build_annotations(self, annotations, url):
         final = []
         if self.annotation_links:
             final.append(url)
+        if not self.annotation_comments:
+            return final
```

Comment:
Kind of nitpicking, but out of an abundance of caution driven by past bugs I've created in methods having multiple (sometimes surprising) `return`s, I'd prefer it if we used this setting to gate whether the loop below is executed rather than returning early.